ads-cert-examples/scala/src/main/scala/Main.scala

```diff
@@ -0,0 +1,88 @@
+import java.io.{FileInputStream, InputStreamReader}
+import java.nio.charset.StandardCharsets
+import java.security.spec.X509EncodedKeySpec
+import java.security._
+
+import org.apache.commons.codec.binary.Base64
+
+import org.bouncycastle.asn1.ASN1ObjectIdentifier
+import org.bouncycastle.asn1.x9.ECNamedCurveTable
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter
+import org.bouncycastle.openssl.{PEMKeyPair, PEMParser}
+
+import scala.util.{Failure, Success, Try}
+
+
+object Main {
+
+  Security.addProvider(new BouncyCastleProvider())
+
+  def main(args: Array[String]): Unit = {
+    val message = Seq(
+      "2824f5d3-a2dc-49ec-8358-dbe92bd01ec2", // source.tid
+      "a:180720yz:1807241326", // ad.pcv
+      "newsite.com", // site.domain
+      "", // app.bundle
+      "", // user.consent
+      "vd", // placement
+      "192.168.1.1", // device.ip
+      "", // device.ipv6
+      "", // device.ifa
+      "", // device.ua
+      "" // ad.video player size?
+    ).mkString(":")
+
+    val tryResult = for {
+      (privateKey, publicKey) <- loadKeys(args)
+      signature <- sign(privateKey, Signature.getInstance("SHA256withECDSA"), message)
+      _ <- Try(println(s"publisher signature: ${Base64.encodeBase64String(signature)}"))
+      result <- verify(publicKey, Signature.getInstance("SHA256withECDSA"), message, signature)
+    } yield result
+
+    tryResult match {
+      case Success(true) => println("yay it worked!")
+      case Success(false) => println("it didn't work :(")
+      case Failure(e) => println(e)
+    }
+  }
+
+  private def loadKeys(args: Array[String]): Try[(PrivateKey, PublicKey)] = {
```

Comment:
For the style you're taking here, you'd want to wrap the method body in a `Try`:

```scala
private def loadKeys(args: Array[String]): Try[(PrivateKey, PublicKey)] = Try {
```

Note the `Try` to the right of the `=`.

Without that thrown `Exception`s aren't wrapped in `Failure` instances. It also means you don't have to explicitly wrap the result at the end of the method in a `Success`; the value that the body evaluates to is successful but any thrown exceptions are failures.

This should keep the most consistency between examples.  Another option is to break down functionality into smaller pieces that are each lifted in to Try contexts either as using `Try { <some Java code> }` or explicit `Success` and `Failure` instances, and then combining those with for-comprehensions.  I think the latter approach would be more commonly preferred -- especially for those with a functional bent, who tend to want to limit Exceptions to as small a scope as possible -- but it's probably not a universal enough style to be considered idiomatic.